lib/metrics/index.js

```diff
@@ -53,6 +53,16 @@ class Metrics {
         return this.serviceName;
     }
 
+    // T247820: Selectively disable service label based on environment variable.
+    // Intended for production use case where the service label will be set
+    // by the Prometheus server.
+    getServiceLabel() {
+        if (process.env.DISABLE_SERVICE_LABEL === 'true') {
```

Comment:
I think its much better to keep switches like this named positivliy.  `disabled = true` is not as understandablle as `enabled = false`.  Can we call this `METRICS_SERVICE_LABEL_ENABLED` and default to true?